docs/developer-guide/etendo-classic/how-to-guides/How_to_implement_a_business_event_handler.md

```diff
@@ -0,0 +1,350 @@
+---
+tags:
+  - HowTo
+  - Etendo Classic
+  - Event Handler
+  - Business Logic
+  - Event Method
+title: How to implement a business event handler 
+---
+  
+## Overview
+
+The  business entity event  allows you to implement business logic which reacts to specific events which are fired when entities are updated, deleted or inserted into the database. 
+Business entity events correspond to triggers in the database. 
+The main advantage of implementing logic using business entity events instead of in triggers is that you can code your logic in java using your IDE. 
+This helps productivity and quality as you can code, debug and test in an integrated environment with the rest of your business logic.
+
+!!! note "Some notes on business entity events:"
+
+  * They are fired when an entity instance is updated, deleted or inserted. Before the actual operation has been done in the database, so you can change or add information which persists together with the event entity. 
+  * Your event handling code runs in the same transaction as the business event, changes you make to the database persist together with the business entity event in one transaction. 
+  * You can make use of the full [data access layer](../concepts/Data_Access_Layer.md) functionality in your event handling code, you can query, create new objects, persist etc. 
```

Comment:
Careful with this, as the DAL shouldn't be used for the target entity ( targetEntity.setX(); OBDal.getInstance().save(targetEntity); targetEntity.getX(); ), instead the setCurrentState / getCurrentState should be used

Reply:
This is stated on the line 188 of this file:
`    // note use setCurrentState and not setters on the Greeting object directly
`
, but I think that a warning in this should be added

Reply:
Warning added